modules/kubernetes-addons/aws-coredns/main.tf

```diff
@@ -159,6 +159,52 @@ resource "null_resource" "modify_kube_dns" {
   }
 }
 
+resource "null_resource" "modify_coredns_metadata" {
+  count = var.enable_self_managed_coredns && var.remove_default_coredns_deployment ? 1 : 0
+
+  triggers = {}
+
+  provisioner "local-exec" {
+    interpreter = ["/bin/bash", "-c"]
+    environment = {
+      KUBECONFIG = base64encode(local.kubeconfig)
+      CONNECTION = null_resource.remove_default_coredns_deployment[0].id
+    }
+
+    # We are maintaining any existing coredns ConfigMap and ServiceAccount,
+    # and annotating and labelling them for Helm to assume control
+    command = <<-EOT
+      _kubectl() {
+        kubectl "$@" \
+          --ignore-not-found \
```

Comment:
I'm not sure if there exist circumstances under which either of the ConfigMap and ServiceAccount would not exist, but I figure it can't hurt to avoid an unnecessary failure.

Reply:
:facepalm: